packages/query/src/block-processor.ts

```diff
@@ -81,27 +81,6 @@ export class BlockProcessor implements BlockProcessorInterface {
     return new TransactionInfo({ height, id, transaction, perspective, view });
   }
 
-  // After a failure, retrying the sync is critical. An exponential-backoff is used.
-  private async autoRetrySync() {
-    try {
-      await backOff(() => this.syncAndStore(), {
-        maxDelay: 30_000, // 30 seconds
-        retry: async error => {
-          await this.viewServer.resetTreeToStored();
-          if (isAbortSignal(error)) return false;
-          console.error('Syncing error', error);
-          return true;
-        },
-      });
-    } catch {
-      // Was signaled to abort, can swallow this error
-    }
-  }
-
-  /**
-   * Populates IndexedDB with data from the blockchain. Most of what's in
-   * IndexedDB comes from this method.
-   */
```

Comment:
now that app parameters can be updated, it's possible for the chainId to change mid-sync. our idb name corresponds to chainId, so if a chainId changes we need to reconnect to another database.

thus, it's important to update the sync error handling in order to react appropriately. i moved retry logic to where it is possible to re-init services when necessary, with the proper idb connection.

formerly, this retry logic called `viewServer.resetTreeToStored`, which cannot complete successfully if the database is closed. the `viewServer` is clobbered if services re-init - it seems like unflushed information might be lost this way, but that can't be fixed without refactoring wasm.

Reply:
I am not sure it's possible for the chain ID to change — I know there are some parameters the governance module doesn't allow changing, maybe the chain ID is one of them. I think it's not essential to handle that possibility at the moment.

Reply:
ok, moved logic back into block-processor where it's simpler. leaving the throw on chainId mismatch because that would indicate some kind of failure